routing/cross_mwm_road_graph.cpp

```diff
@@ -5,9 +5,41 @@
 
 namespace
 {
+using namespace routing;
+
 inline bool IsValidEdgeWeight(EdgeWeight const & w) { return w != INVALID_EDGE_WEIGHT; }
+
+void FindIngoingCrossNode(CrossRoutingContextReader const & currentContext, CrossNode const & toNode,
+                          IngoingCrossNode & ingoingNode)
+{
+  bool found = false;
+  auto const findingFn = [&ingoingNode, &toNode, &found](IngoingCrossNode const & node){
+    if (node.m_nodeId == toNode.node)
```

Comment:
It happens sometimes. In that case an random outgoing/ingoing node will be taken. It leads to a well known issue: jumping from one road to a wrong one on mwm border.

Here we can write a work around and fix issues.
Instead of a random node to take a closest to CrossNode one. I'll to that.